### ISSUE TITLE
[GLPI 10.0.3] SQL errors on search elements linked to a location

### ISSUE THREAD

**Issue by tsmr** [glpi-project/glpi#12797]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

[GLPI 10 Bugfixes]

### Bug description

Go on a location, Elements tab

```
[2022-09-28 07:13:29] glpisqllog.ERROR: DBmysql::query() in C:\wamp64\www\glpi\src\DBmysql.php line 370
  *** MySQL query error:
  SQL: SELECT * FROM ((SELECT `glpi_budgets`.`id`, 'Budget' AS `type` FROM `glpi_budgets` WHERE `glpi_budgets`.`locations_id` = '168' AND (`glpi_budgets`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_cartridgeitems`.`id`, 'CartridgeItem' AS `type` FROM `glpi_cartridgeitems` WHERE `glpi_cartridgeitems`.`locations_id` = '168' AND (`glpi_cartridgeitems`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_consumableitems`.`id`, 'ConsumableItem' AS `type` FROM `glpi_consumableitems` WHERE `glpi_consumableitems`.`locations_id` = '168' AND (`glpi_consumableitems`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_computers`.`id`, 'Computer' AS `type` FROM `glpi_computers` WHERE `glpi_computers`.`locations_id` = '168' AND (`glpi_computers`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_monitors`.`id`, 'Monitor' AS `type` FROM `glpi_monitors` WHERE `glpi_monitors`.`locations_id` = '168' AND (`glpi_monitors`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_sockets`.`id`, 'Glpi\\Socket' AS `type` FROM `glpi_sockets` WHERE `glpi_sockets`.`locations_id` = '168' AND (`glpi_sockets`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244'))) UNION ALL (SELECT `glpi_networkequipments`.`id`, 'NetworkEquipment' AS `type` FROM `glpi_networkequipments` WHERE `glpi_networkequipments`.`locations_id` = '168' AND (`glpi_networkequipments`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_peripherals`.`id`, 'Peripheral' AS `type` FROM `glpi_peripherals` WHERE `glpi_peripherals`.`locations_id` = '168' AND (`glpi_peripherals`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_phones`.`id`, 'Phone' AS `type` FROM `glpi_phones` WHERE `glpi_phones`.`locations_id` = '168' AND (`glpi_phones`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_printers`.`id`, 'Printer' AS `type` FROM `glpi_printers` WHERE `glpi_printers`.`locations_id` = '168' AND (`glpi_printers`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_softwares`.`id`, 'Software' AS `type` FROM `glpi_softwares` WHERE `glpi_softwares`.`locations_id` = '168' AND (`glpi_softwares`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_softwarelicenses`.`id`, 'SoftwareLicense' AS `type` FROM `glpi_softwarelicenses` WHERE `glpi_softwarelicenses`.`locations_id` = '168' AND (`glpi_softwarelicenses`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_tickets`.`id`, 'Ticket' AS `type` FROM `glpi_tickets` WHERE `glpi_tickets`.`locations_id` = '168' AND (`glpi_tickets`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_users`.`id`, 'User' AS `type` FROM `glpi_users` WHERE `glpi_users`.`locations_id` = '168' AND (`glpi_users`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_certificates`.`id`, 'Certificate' AS `type` FROM `glpi_certificates` WHERE `glpi_certificates`.`locations_id` = '168' AND (`glpi_certificates`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicesimcards`.`id`, 'Item_DeviceSimcard' AS `type` FROM `glpi_items_devicesimcards` WHERE `glpi_items_devicesimcards`.`locations_id` = '168' AND (`glpi_items_devicesimcards`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_lines`.`id`, 'Line' AS `type` FROM `glpi_lines` WHERE `glpi_lines`.`locations_id` = '168' AND (`glpi_lines`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_appliances`.`id`, 'Appliance' AS `type` FROM `glpi_appliances` WHERE `glpi_appliances`.`locations_id` = '168' AND (`glpi_appliances`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_passivedcequipments`.`id`, 'PassiveDCEquipment' AS `type` FROM `glpi_passivedcequipments` WHERE `glpi_passivedcequipments`.`locations_id` = '168' AND (`glpi_passivedcequipments`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_datacenters`.`id`, 'DataCenter' AS `type` FROM `glpi_datacenters` WHERE `glpi_datacenters`.`locations_id` = '168' AND (`glpi_datacenters`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_dcrooms`.`id`, 'DCRoom' AS `type` FROM `glpi_dcrooms` WHERE `glpi_dcrooms`.`locations_id` = '168' AND (`glpi_dcrooms`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_racks`.`id`, 'Rack' AS `type` FROM `glpi_racks` WHERE `glpi_racks`.`locations_id` = '168' AND (`glpi_racks`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_enclosures`.`id`, 'Enclosure' AS `type` FROM `glpi_enclosures` WHERE `glpi_enclosures`.`locations_id` = '168' AND (`glpi_enclosures`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_pdus`.`id`, 'PDU' AS `type` FROM `glpi_pdus` WHERE `glpi_pdus`.`locations_id` = '168' AND (`glpi_pdus`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicemotherboards`.`id`, 'Item_DeviceMotherboard' AS `type` FROM `glpi_items_devicemotherboards` WHERE `glpi_items_devicemotherboards`.`locations_id` = '168' AND (`glpi_items_devicemotherboards`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicefirmwares`.`id`, 'Item_DeviceFirmware' AS `type` FROM `glpi_items_devicefirmwares` WHERE `glpi_items_devicefirmwares`.`locations_id` = '168' AND (`glpi_items_devicefirmwares`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_deviceprocessors`.`id`, 'Item_DeviceProcessor' AS `type` FROM `glpi_items_deviceprocessors` WHERE `glpi_items_deviceprocessors`.`locations_id` = '168' AND (`glpi_items_deviceprocessors`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicememories`.`id`, 'Item_DeviceMemory' AS `type` FROM `glpi_items_devicememories` WHERE `glpi_items_devicememories`.`locations_id` = '168' AND (`glpi_items_devicememories`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_deviceharddrives`.`id`, 'Item_DeviceHardDrive' AS `type` FROM `glpi_items_deviceharddrives` WHERE `glpi_items_deviceharddrives`.`locations_id` = '168' AND (`glpi_items_deviceharddrives`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicenetworkcards`.`id`, 'Item_DeviceNetworkCard' AS `type` FROM `glpi_items_devicenetworkcards` WHERE `glpi_items_devicenetworkcards`.`locations_id` = '168' AND (`glpi_items_devicenetworkcards`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicedrives`.`id`, 'Item_DeviceDrive' AS `type` FROM `glpi_items_devicedrives` WHERE `glpi_items_devicedrives`.`locations_id` = '168' AND (`glpi_items_devicedrives`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicebatteries`.`id`, 'Item_DeviceBattery' AS `type` FROM `glpi_items_devicebatteries` WHERE `glpi_items_devicebatteries`.`locations_id` = '168' AND (`glpi_items_devicebatteries`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicegraphiccards`.`id`, 'Item_DeviceGraphicCard' AS `type` FROM `glpi_items_devicegraphiccards` WHERE `glpi_items_devicegraphiccards`.`locations_id` = '168' AND (`glpi_items_devicegraphiccards`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicesoundcards`.`id`, 'Item_DeviceSoundCard' AS `type` FROM `glpi_items_devicesoundcards` WHERE `glpi_items_devicesoundcards`.`locations_id` = '168' AND (`glpi_items_devicesoundcards`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicecontrols`.`id`, 'Item_DeviceControl' AS `type` FROM `glpi_items_devicecontrols` WHERE `glpi_items_devicecontrols`.`locations_id` = '168' AND (`glpi_items_devicecontrols`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicepcis`.`id`, 'Item_DevicePci' AS `type` FROM `glpi_items_devicepcis` WHERE `glpi_items_devicepcis`.`locations_id` = '168' AND (`glpi_items_devicepcis`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicecases`.`id`, 'Item_DeviceCase' AS `type` FROM `glpi_items_devicecases` WHERE `glpi_items_devicecases`.`locations_id` = '168' AND (`glpi_items_devicecases`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicepowersupplies`.`id`, 'Item_DevicePowerSupply' AS `type` FROM `glpi_items_devicepowersupplies` WHERE `glpi_items_devicepowersupplies`.`locations_id` = '168' AND (`glpi_items_devicepowersupplies`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicegenerics`.`id`, 'Item_DeviceGeneric' AS `type` FROM `glpi_items_devicegenerics` WHERE `glpi_items_devicegenerics`.`locations_id` = '168' AND (`glpi_items_devicegenerics`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicesimcards`.`id`, 'Item_DeviceSimcard' AS `type` FROM `glpi_items_devicesimcards` WHERE `glpi_items_devicesimcards`.`locations_id` = '168' AND (`glpi_items_devicesimcards`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0') UNION ALL (SELECT `glpi_items_devicesensors`.`id`, 'Item_DeviceSensor' AS `type` FROM `glpi_items_devicesensors` WHERE `glpi_items_devicesensors`.`locations_id` = '168' AND (`glpi_items_devicesensors`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244')) AND `is_deleted` = '0')) AS `union_4b23637ff24d00145142b821fdf2c6d1` LIMIT 200
  Error: Champ 'glpi_sockets.entities_id' inconnu dans where clause
  Backtrace :
  src\DBmysqlIterator.php:110                        
  src\DBmysql.php:1048                               DBmysqlIterator->execute()
  src\Location.php:450                               DBmysql->request()
  src\Location.php:395                               Location->showItems()
  src\CommonGLPI.php:689                             Location::displayTabContentForItem()
  ajax\common.tabs.php:113                           CommonGLPI::displayStandardTab()
```

**Comment by stonebuzz** [comment]
Hi @tsmr 
can you try this -> https://github.com/glpi-project/glpi/pull/12798

Best regards

**Comment by cedric-anne** [comment]
Fixed by #12798
